### PR TITLE
SASS-9572: Foreign property - 'Select country' section:  Implement DES/IF Create & Update API

### PR DESCRIPTION
**Pull request by vivekhmrc**

### CODE REVIEW

File: app/controllers/JourneyAnswersController.scala [hunk 1]
```diff
@@ -358,4 +359,14 @@ class JourneyAnswersController @Inject() (
           }
       }
     }
+
+  def saveForeignPropertiesInformation(taxYear: TaxYear, incomeSourceId: IncomeSourceId, nino: Nino): Action[AnyContent] =
```
Comment (arunkumar461) [inline app/controllers/JourneyAnswersController.scala:363]: Move it to new controller?

Comment (arunkumar461) [reply app/controllers/JourneyAnswersController.scala:363]: saveForeignPropertiesInformation --> saveSelectCountrySection ?

File: app/models/common/JourneyName.scala [hunk 2]
```diff
@@ -70,4 +70,5 @@ object JourneyName extends Enum[JourneyName] with utils.PlayJsonEnum[JourneyName
 
   case object RentalsAndRaRESBA extends JourneyName("property-rentals-and-rent-a-room-esba")
 
+  case object ForeignPropertySelectCountries extends JourneyName("foreign-property-select-countries")
```
Comment (arunkumar461) [inline app/models/common/JourneyName.scala:73]: select-countries --> select-country

File: app/models/request/foreign/ForeignPropertiesInformation.scala [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+import play.api.libs.json.{Json, OFormat}
+
+final case class ForeignPropertiesInformation(
+  countryCodes: List[String],
+  foreignTotalIncome: String,
+  claimPropertyIncomeAllowanceOrExpenses: String
+)
+
+object ForeignPropertiesInformation {
```
Comment (arunkumar461) [inline app/models/request/foreign/ForeignPropertiesInformation.scala:27]: Section name would be good for identification of the class ?

File: app/services/PropertyService.scala [hunk 4]
```diff
@@ -662,4 +663,18 @@ class PropertyService @Inject() (
       res <- persistAnswers(ctx.toJourneyContext(JourneyName.RentARoomAllowances), rentARoomAllowancesStoreAnswers)
     } yield res
   }
+
+  def saveForeignPropertiesInformation(
+    ctx: JourneyContext,
+    nino: Nino,
+    foreignPropertiesInformation: ForeignPropertiesInformation
+  )(implicit hc: HeaderCarrier): EitherT[Future, ServiceError, Boolean] =
```
Comment (arunkumar461) [inline app/services/PropertyService.scala:671]: I think we don't need of EitherT here.

**Review by arunkumar461 — CHANGES_REQUESTED** [review]

File: app/models/common/JourneyName.scala (hunk 2 above)
Comment (vivekhmrc) [reply app/models/common/JourneyName.scala]: Changed

File: app/controllers/JourneyAnswersController.scala (hunk 1 above)
Comment (vivekhmrc) [reply app/controllers/JourneyAnswersController.scala]: Renamed.

File: app/services/PropertyService.scala (hunk 4 above)
Comment (vivekhmrc) [reply app/services/PropertyService.scala:671]: We don't need a for comprehension but the persistAnswers type is EitherT

**Review by arunkumar461 — APPROVED** [review]

**Pull request merged**